src/ansys/fluent/core/session_utilities.py

```diff
@@ -0,0 +1,426 @@
+        gpu : bool, optional
+            If True, starts Fluent with GPU Solver enabled.
+        start_watchdog : bool, optional
+            If True and `cleanup_on_exit` is True, an independent watchdog process is run to ensure that any local
+            GUI-less Fluent sessions started by PyFluent are properly closed when the current Python process ends.
+        file_transfer_service : Any, optional
+            Service for uploading/downloading files to/from the server.
+
+        Returns
+        -------
+        Union[Meshing, PureMeshing, Solver, SolverIcing, dict]
+            Session object or configuration dictionary if ``dry_run`` is True.
+
+        Raises
+        ------
+        UnexpectedKeywordArgument
+            If an unexpected keyword argument is provided.
+
+        Notes
+        -----
+        In job scheduler environments (e.g., SLURM, LSF, PBS), resources and compute nodes are allocated,
+        and core counts are queried from these environments before being passed to Fluent.
+        """
+        mode = cls._session_mode[cls.__name__]
+        argvals = locals().copy()
+        argvals.pop("cls", None)
+        launcher = PIMLauncher(**argvals)
+        return launcher()
+
+    @classmethod
+    def from_connection(
+        cls,
+        ip: str | None = None,
+        port: int | None = None,
+        server_info_file_name: str | None = None,
+        password: str | None = None,
+    ):
+        """Connect to an existing Fluent server instance.
+
+        Parameters
+        ----------
+        ip : str, optional
+            IP address for connecting to an existing Fluent instance. The
+            IP address defaults to ``"127.0.0.1"``. You can also use the environment
+            variable ``PYFLUENT_FLUENT_IP=<ip>`` to set this parameter.
+            The explicit value of ``ip`` takes precedence over ``PYFLUENT_FLUENT_IP=<ip>``.
+        port : int, optional
+            Port to listen on for an existing Fluent instance. You can use the
+            environment variable ``PYFLUENT_FLUENT_PORT=<port>`` to set a default
+            value. The explicit value of ``port`` takes precedence over
+            ``PYFLUENT_FLUENT_PORT=<port>``.
+        server_info_file_name: str
+            Path to server-info file written out by Fluent server. The default is
+            ``None``. PyFluent uses the connection information in the file to
+            connect to a running Fluent session.
+        password : str, optional
+            Password to connect to existing Fluent instance.
+
+        Raises
+        ------
+        TypeError
+            If the session type does not match the expected session type.
+        """
+        session = pyfluent.connect_to_fluent(
+            ip=ip,
+            port=port,
+            server_info_file_name=server_info_file_name,
+            password=password,
+        )
+
+        if session.__class__.__name__ != cls.__name__:
```

Comment:
These names should exactly match the class names. There might be an issue here, though a better logic is not coming to my mind as well now. Like for example, we will never be able to connect to a PrePost session using our current PyFluent.
If others have a better design in mind lease comment.
Thank you.

Reply:
Yes, we need a class for PrePost session similar to existing session types.

Reply:
@prmukherj We have handled this scenario now. No need to have a separate PrePost class.

Reply:
Looks good for the particular scenario.